contracts/ColonyRoles.sol

```diff
@@ -0,0 +1,85 @@
+/*
+  This file is part of The Colony Network.
+
+  The Colony Network is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity >=0.4.23;
+
+import "./CommonAuthority.sol";
+
+
+contract ColonyRoles is CommonAuthority {
```

Comment:
I don't understand why this implementation is needed. The `DSRoles` vanilla contract allows for up to 256 roles to be mapped to essentially 63 uniquely permissioned domains (with 4 unique roles each), which I imagine is sufficiently high limit considering we inherit permissions by default.

Reply:
Mm I think it's quite risky to have an implicit cap on domains -- I can imagine many colonies who will experiment with domain structures and then "run out" of domains. Imagine a large colony who wants to grow but then realizes it's "stuck". That seems like a bad outcome. And we can have unlimited domains with this implementation, and it's not like it's a huge additional cost (esp. since there's no cost for a domain with no explicit permissions).